mmaction/datasets/base.py

```diff
@@ -85,19 +87,72 @@ def load_json_annotations(self):
                 video_infos[i]['label'] = video_infos[i]['label'][0]
         return video_infos
 
-    @abstractmethod
-    def evaluate(self, results, metrics, logger):
-        """Evaluation for the dataset.
+    def evaluate(self,
+                 results,
+                 metrics='top_k_accuracy',
+                 metric_dict=dict(topk=(1, 5)),
+                 logger=None):
+        """Evaluation in rawframe dataset.
 
         Args:
             results (list): Output results.
             metrics (str | sequence[str]): Metrics to be performed.
+                Defaults: 'top_k_accuracy'.
+            logger (obj): Training logger. Defaults: None.
+            metric_dict (dict): Dict for metric options.
+                Default: ``dict(topk=(1, 5))``.
             logger (logging.Logger | None): Logger for recording.
+                Default: None.
 
         Returns:
             dict: Evaluation results dict.
         """
-        pass
+        if not isinstance(results, list):
+            raise TypeError(f'results must be a list, but got {type(results)}')
+        assert len(results) == len(self), (
+            f'The length of results is not equal to the dataset len: '
+            f'{len(results)} != {len(self)}')
+
+        metrics = metrics if isinstance(metrics, (list, tuple)) else [metrics]
```

Comment:
Need mean average precision here.

Reply:
I have noticed that other datasets, namely rawframe_dataset, video_dataset have override it by add the mean_average_precision, why not move it into Base and don't override since they are the same.

Reply:
done.